### ISSUE TITLE
add fine control of which latent variables to update in update()

### ISSUE THREAD

**Issue by dustinvtran** [blei-lab/edward#334]
there should be an argument `latent_vars` to `update()`, which is a list of the latent variables to update. by default, `update()` updates all of the latent variables.
+ for example, this enables many updates of local latents before then updating global variables
+ we can implement this by teasing out the train op in variational inference. update will only run the train op for all tensorflow variables that the chosen latent variables depend on
+ for mcmc, this is easy because the train/assign op simply does not need to be grouped any longer.

**Comment by dustinvtran** [comment]
In general, is this something that should be done over just defining inference as a collection of many separate inferences? I'm not sure. It's clear the latter is more general. I think the former is oftentimes more convenient, and to do it well, we have to be clear on what restrictions it imposes.

more notes
+ this requires an optimizer for each latent variable. the `optimizer` argument is generally a dictionary of optimizers. (that's a bit confusing)
+ how to deal with parameters that are used among latent variables? most generally, it sounds like we need individual optimizers for each __parameter__.
+ this is also related to using different gradients for each latent var (#66)

**Comment by dustinvtran** [comment]
@ajbc asks:

> Say I want to do stochastic inference and I have local variables that I don’t need updated every iteration—or rather, I only want to update a subset of them, how do I structure that in edward?  Is it possible currently?

I think the easiest way is to instantiate two local inferences, one for the subset you want updated every iteration and another for the subset you want to update only at specific iterations.

More work needs to be done to make these possible in just one inference.
